File: python/Binary_Tree_Level_Order_Traversal.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param root, a tree node
    # @return a list of lists of integers
    def levelOrder(self, root):
        lists=[]
        if root is None:
            return lists
        queue1=[]
        queue2=[]
        queue1.append(root)
        while len(queue1) is not 0:
            list=[]
            for i in queue1:
                list.append(i.val);
                if i.left is not None:
                    queue2.append(i.left)
                if i.right is not None:
                    queue2.append(i.right)
            lists.append(list)
            list=[]
            queue1=[]
            if len(queue2) is not 0:
                for i in queue2:
                    list.append(i.val)
                    if i.left is not None:
                        queue1.append(i.left)
                    if i.right is not None:
                        queue1.append(i.right)
                lists.append(list)
                list=[]
                queue2=[]
        return lists

File: c++/Binary_Tree_Level_Order_Traversal.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<vector<int> > levelOrder(TreeNode *root) {
        vector<vector<int>> ret;
        if(root==NULL) {
            return ret;
        }
        list<TreeNode *> level[2];
        level[0].push_back(root);
        while(level[0].size()||level[1].size()) {
            vector<int> levelList;
            int i;
            if(level[0].size()) {
                i=0;
            } else {
                i=1;
            }
            while(level[i].size()) {
                TreeNode *node=level[i].front();
                level[i].pop_front();
                levelList.push_back(node->val);
                if(node->left) {
                    level[(i+1)%2].push_back(node->left);
                }
                if(node->right) {
                    level[(i+1)%2].push_back(node->right);
                }
            }
            ret.push_back(levelList);
        }
        return ret;
    }
};